### PR TITLE
Upgrade MicrosoftTranslate to v3 API

### PR DESCRIPTION
**Pull request by miaocup**
> On 24 May 2021, Translator version 2 will retire and stop working.

_(:з」∠)_

### CODE REVIEW

File: ACT.FFXIVTranslate/ACT.FFXIVTranslate/translate/microsoft/MicrosoftTranslateProvider.cs [hunk 1]
```diff
@@ -148,18 +130,20 @@ private string GetAuthToken()
...
         }
 
-        private void DoRequest(string authToken, string requestBody,
+        private void DoRequest(string authToken, string uri, string requestBody,
             out string responseBody, out HttpStatusCode statusCode)
         {
             string _responseBody;
             HttpStatusCode _statusCode;
 
+            ServicePointManager.SecurityProtocol |= SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;
```
Comment (Noisyfox) [inline ACT.FFXIVTranslate/ACT.FFXIVTranslate/translate/microsoft/MicrosoftTranslateProvider.cs:139]: Is there a good reason to allow TLSv1.1 here?

**Review by Noisyfox — CHANGES_REQUESTED** [review]

File: ACT.FFXIVTranslate/ACT.FFXIVTranslate/translate/microsoft/MicrosoftTranslateProvider.cs (hunk 1 above)
Comment (miaocup) [reply ACT.FFXIVTranslate/ACT.FFXIVTranslate/translate/microsoft/MicrosoftTranslateProvider.cs:139]: It seems the API endpoint does not support the default one. 
> "connection force closed by remote". 
Resolved after adding this line. Not sure if there is a better way.

Comment (Noisyfox) [reply ACT.FFXIVTranslate/ACT.FFXIVTranslate/translate/microsoft/MicrosoftTranslateProvider.cs:139]: Interesting... I'd imagine it should support up to tlsv1.3. Anyway this looks fine to me.

**Comment by Noisyfox** [comment]
Thanks~

**Pull request merged**